### PR TITLE
Update Popup explainer to incorporate all recent changes

### PR DESCRIPTION
**Pull request by mfreed7**
This PR also revamps the explainer significantly, changing the language from "proposal" to "explainer". Many details are also added, such as what "light dismiss" really means, the "nearest open ancestral popup" algorithm, and many more details. I also added links back to many of the recent resolutions where some of these decisions made.

I'm sure there are typos and issues here, but I did a first editing pass. Comments appreciated.

### CODE REVIEW

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (andrico1234) [inline research/src/pages/popup/popup.research.explainer.mdx:487]: Unless I've made a silly mistake, won't the following: 

`While the [original `<popup>` proposal](https://open-ui.org/components/popup.research.explainer#goals)...`

just link to the current page's goals heading? 

Is there a different document that this link should point to, or is the wording just a little ambiguous?

I'd figure that the _original <popup> proposal_ would be an older spec (maybe the abandoned one?)

**Comment by andrico1234** [comment]
I did a one-pass of the preview site, and the document looks great and well-formatted!

Happy to approve, but not sure if you want someone more familiar with the spec to review the content :)

**Review by andrico1234 — APPROVED** [review]

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (scottaohara) [inline research/src/pages/popup/popup.research.explainer.mdx:184]:
```suggestion
The above rules mean that a popup, when not "shown", has `display:none` applied, and that style is removed when one of the methods above is used to show the popup. Note that the `display:none` UA stylesheet rule is **not** `!important`. In other words, developer style rules can be used to override this UA style to make a not-showing popup visible in the page. In this case, the popup will **not** be displayed in the top layer, but rather at it's ordinary `z-index` position within the document. This can be used, for example, to animate the show/hide behavior of the popup, or make popup content "return to the page" instead of becoming hidden.
```

**Review by scottaohara — COMMENTED** [review]
was not able to get through the whole doc, but two things for now.

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (mfreed7) [reply research/src/pages/popup/popup.research.explainer.mdx]: Yep, done, thanks!

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (mfreed7) [reply research/src/pages/popup/popup.research.explainer.mdx]: Great catch! That's from before I moved the explainer to here.

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (mfreed7) [reply research/src/pages/popup/popup.research.explainer.mdx]: Thanks! done.

**Comment by mfreed7** [comment]
Thanks for the reviews. @scottaohara let me know if you want me to wait for you to take a look at the rest before I land it.

**Comment by scottaohara** [comment]
going to finish up now.

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (scottaohara) [inline research/src/pages/popup/popup.research.explainer.mdx:236]: note that this use of `autofocus` is also relevant to the initial focus placement doc/wiki that was worked on for the dialog element.  so there is pending precedent for this proposal for the attribute.

```suggestion
- `autofocus`. When present on a popup or one of its descendants, it will result in focus being moved to the specified element when the popup is rendered. Note that `autofocus` is [already a global attribute](https://html.spec.whatwg.org/multipage/interaction.html#the-autofocus-attribute), but the existing behavior applies to element focus on **page load**. This proposal extends that definition to be used within popups, and the focus behavior happens **when they are shown**. Note that adding `autofocus` to a popup descendant does **not** cause the popup to be shown on page load, and therefore it does not cause focus to be moved into the popup **on page load**, unless the `defaultopen` attribute is also used.
```

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (scottaohara) [inline research/src/pages/popup/popup.research.explainer.mdx:238]: Would it be good to call out that the purpose of such an attribute would be for instances where the subtree of the popup may not be known prior to its rendering?  Otherwise, it seems like something that specifying `autofocus` on the first focusable element of a popup could have taken care of?

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (scottaohara) [inline research/src/pages/popup/popup.research.explainer.mdx:275]: just a consistency nit, putting the period inside the bold like the previous

```suggestion
* **Light Dismiss.** User action such as clicking outside the element, hitting Escape, or causing keyboard focus to leave the element should all cause a displayed popup to be hidden. This is typically called “light dismiss”, and is discussed in more detail in [this section](#light-dismiss).
```

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (scottaohara) [inline research/src/pages/popup/popup.research.explainer.mdx:276]: same same
```suggestion
* **Other Reasons.** Because the top layer is a UA-managed resource, it may have other reasons (for example a user preference) to forcibly remove elements from the top layer.
```

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (scottaohara) [inline research/src/pages/popup/popup.research.explainer.mdx:284]: suggestion, maybe `mouseup` would be good here?  

I say this because [WCAG 2.5.2 Pointer Cancellation criterion](https://www.w3.org/WAI/WCAG21/Understanding/pointer-cancellation.html) could then be adhered to, allowing someone who makes an accidental down click outside of a popup (someone with motor disabilities, for instance) would have the chance to move their pointer back into the boundaries of the popup, allowing for the accidental dismissal to be aborted.

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (scottaohara) [inline research/src/pages/popup/popup.research.explainer.mdx:285]: matching earlier use of the full word, rather than the abbreviated key.

```suggestion
2. Hitting the Escape key, or generally any ["close signal"](#close-signal). This will cause the topmost popup on [the popup stack](#the-popup-stack) to be hidden.
```

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (scottaohara) [inline research/src/pages/popup/popup.research.explainer.mdx:373]:
```suggestion
This section details the interactions between the three popup types:
```

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (scottaohara) [inline research/src/pages/popup/popup.research.explainer.mdx:389]: Swapping around the examples a bit because while I still think they should be valid, using an image as a popup could definitely have its use cases (i'm immediately thinking of "here's where your credit card code is"), but not the common use case I'd imagine people would go for.  

```suggestion
Since the `popup` content attribute can be applied to any element, and this only impacts the element's presentation (top layer vs not top layer), this addition does not have any direct semantic or accessibility impact. The element with the `popup` attribute will keep its existing semantics and AOM representation. For example, `<article popup=popup>...</article>` will continue to be exposed as an implicit `role=article`, but will be able to be displayed on top of other content. Similarly, ARIA can be used to modify accessibility mappings in [the normal way](https://w3c.github.io/html-aria/), for example `<div popup=popup role=note>...</div>`.
```

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (scottaohara) [inline research/src/pages/popup/popup.research.explainer.mdx:395]: I do not think a dialog should be completely barred from being a popup (unless i'm misunderstanding this line?)  it's also in opposition with the example that follows in the next section, re: `<my-date-picker role=dialog id=datepicker popup=popup>`

can expand on this if need be... but maybe this should just be clarified to mean a 'modal' dialog would throw an exception?  (even though there are tons of instances of custom modal dialogs where people implement a click-outside-to-close-light-dismiss... but, i have too many other hills i've decided to die on...)

**Review by scottaohara — COMMENTED** [review]
OK, have finished reviewing.  Thank you for all the work on this.  Please let me know if I need to add further clarification to any of my comments / suggestions

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (mfreed7) [reply research/src/pages/popup/popup.research.explainer.mdx]: I've gone back and forth on this. But I do see your point here that `role=dialog` semantics might be entirely appropriate on something that has popup behaviors (like click-outside light dismiss).

I've added explicitly "modal" dialog here.

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (mfreed7) [reply research/src/pages/popup/popup.research.explainer.mdx]: Awesome, thanks. I struggled with a good example here - this is better.

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (mfreed7) [reply research/src/pages/popup/popup.research.explainer.mdx]: Nice

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (mfreed7) [reply research/src/pages/popup/popup.research.explainer.mdx]: Done

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (mfreed7) [reply research/src/pages/popup/popup.research.explainer.mdx:284]: This is likely worth discussion. I did a rather informal survey of native and web popups, and found that nearly all of them dismiss on mouse-down. Think it's worth an issue?

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (mfreed7) [reply research/src/pages/popup/popup.research.explainer.mdx]: Thanks! I hate when those aren't consistent.

On the `displayed` vs `showing`, I fought with that throughout. I wanted to avoid "open"/"closed" because we're using "show"/"hide", but it's awkward in many cases. I like "displayed", I might try to use that more.

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (mfreed7) [reply research/src/pages/popup/popup.research.explainer.mdx]: Yep, good point. Done.

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (mfreed7) [reply research/src/pages/popup/popup.research.explainer.mdx]: Accepted.

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (mfreed7) [reply research/src/pages/popup/popup.research.explainer.mdx]: I like it. Not sure why the diff is terrible, but I got it.

**Comment by mfreed7** [comment]
> OK, have finished reviewing. Thank you for all the work on this. Please let me know if I need to add further clarification to any of my comments / suggestions

Thanks for the great review! I resolved everything except one question. It might warrant a new issue to discuss further.

**Comment by mfreed7** [comment]
I'm going to land this now, but let me know what you think about the mouse-down vs. mouse-up question and I can open an issue for that.

File: research/src/pages/popup/popup.research.explainer.mdx
Comment (scottaohara) [reply research/src/pages/popup/popup.research.explainer.mdx:284]: i think it's definitely worth at least a discussion.

there are presently a lot of platform behaviors that get a 'pass' with wcag rules because web authors can literally do nothing about them, since they're baked into the platform.  but, if given the chance to change a behavior, it at least deserves being brought up and then further discussion can dictate if there's a reason to adhere to what's been previously done vs going in a different direction.  

thanks!

Comment (mfreed7) [reply research/src/pages/popup/popup.research.explainer.mdx:284]: Filed! https://github.com/openui/open-ui/issues/529

**Pull request merged**
